tests/common/features/soft_delete_many_to_many.rs

```diff
@@ -0,0 +1,742 @@
+}
+
+pub mod child_with_sd_via_sd_junction {
+    use super::junction_with_soft_delete;
+    use super::parent;
+    use sea_orm::entity::prelude::*;
+
+    #[derive(Clone, Debug, PartialEq, DeriveEntityModel)]
+    #[sea_orm(table_name = "soft_delete_child")]
+    pub struct Model {
+        #[sea_orm(primary_key)]
+        pub id: i32,
+        pub parent_id: i32,
+        pub name: String,
+        pub created_at: Option<DateTime>,
+        pub updated_at: Option<DateTime>,
+        #[sea_orm(soft_delete_column)]
+        pub deleted_at: Option<DateTime>,
+    }
+
+    #[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
+    pub enum Relation {}
+
+    impl Related<parent::Entity> for Entity {
+        fn to() -> RelationDef {
+            junction_with_soft_delete::Relation::Parent.def()
+        }
+
+        fn via() -> Option<RelationDef> {
+            Some(
+                junction_with_soft_delete::Relation::SoftDeleteChildViaSoftDeleteJunction
+                    .def()
+                    .rev(),
+            )
+        }
+    }
+
+    pub struct SoftDeleteChildToParent;
+
+    impl Linked for SoftDeleteChildToParent {
+        type FromEntity = Entity;
+
+        type ToEntity = parent::Entity;
+
+        fn link(&self) -> Vec<RelationDef> {
+            vec![
+                junction_with_soft_delete::Relation::SoftDeleteChildViaSoftDeleteJunction
+                    .def()
+                    .rev(),
+                junction_with_soft_delete::Relation::Parent.def(),
+            ]
+        }
+    }
+
+    impl ActiveModelBehavior for ActiveModel {}
+
+    #[cfg(test)]
+    mod tests {
+        use super::*;
+        use pretty_assertions::assert_eq;
+        use sea_orm::*;
+
+        #[test]
+        fn find_related_eager() {
+            let find_parent: Select<parent::Entity> = Entity::find_related();
+            assert_eq!(
+                find_parent
+                    .filter(Column::Id.eq(11))
+                    .build(DbBackend::MySql)
+                    .to_string(),
+                [
+                    "SELECT `soft_delete`.`id`, `soft_delete`.`name`, `soft_delete`.`created_at`, `soft_delete`.`updated_at`, `soft_delete`.`deleted_at`",
+                    "FROM `soft_delete`",
+                    "INNER JOIN `soft_delete_junction` ON `soft_delete_junction`.`parent_id` = `soft_delete`.`id`",
+                    "INNER JOIN `soft_delete_child` ON `soft_delete_child`.`id` = `soft_delete_junction`.`child_id`",
+                    "WHERE `soft_delete`.`deleted_at` IS NULL",
+                    // FIXME: No way to know if the junction have soft delete enabled with only RelationDef on hand
+                    // "AND `soft_delete_junction`.`deleted_at` IS NULL",
+                    "AND `soft_delete_child`.`deleted_at` IS NULL",
+                    "AND `soft_delete_child`.`id` = 11",
+                ]
+                .join(" ")
+            );
+        }
```

Comment:
Currently, we have no way to tell if the junction table have soft delete enabled or not with `Related<T>`.